### PR TITLE
Fixed ISE when fragment is not attached to activity

### LINKED ISSUE #3743: IllegalStateException: Fragment d{2bc4a5b} (0c1c916c-2785-488a-b67c-1c7f27f453f0)} not attached to an activity.
Sentry Issue: [OPENFOODFACTS-ANDROID-1RB](https://sentry.io/organizations/openfoodfacts/issues/2139235792/?referrer=github_integration)

```
IllegalStateException: Fragment d{2bc4a5b} (0c1c916c-2785-488a-b67c-1c7f27f453f0)} not attached to an activity.
    at androidx.fragment.app.Fragment.requireActivity(Fragment.java:926)
    at openfoodfacts.github.scrachx.openfood.features.product.view.ingredients.IngredientsProductFragment.extractIngredients(IngredientsProductFragment.kt:381)
    at openfoodfacts.github.scrachx.openfood.features.product.view.ProductViewFragment.showIngredientsTab(ProductViewFragment.kt:120)
    at openfoodfacts.github.scrachx.openfood.features.scan.ContinuousScanActivity.showIngredientsTab(ContinuousScanActivity.kt:696)
    at openfoodfacts.github.scrachx.openfood.features.product.view.ingredients_analysis.IngredientsWithTagDialogFragment.goToExtract(IngredientsWithTagDialogFragment.kt:144)
...
(13 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by mrudultora**
####  Description
Used `isAdded()` to check whether the fragment is attached to an activity.

#### Related issues
Fixes - #3743 

#### Related PRs
None

#### Screenshots

### CODE REVIEW

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/features/product/view/ingredients/IngredientsProductFragment.kt [hunk 1]
```diff
@@ -379,12 +379,14 @@ class IngredientsProductFragment : BaseFragment(), IIngredientsProductPresenter.
 
     fun extractIngredients() {
         ingredientExtracted = true
-        val settings = requireActivity().getLoginPreferences()
-        if (settings.getString("user", "")!!.isEmpty()) {
-            showSignInDialog()
-        } else {
-            productState = requireProductState()
-            performOCRLauncher.launch(productState.product)
+        if (isAdded) {
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/features/product/view/ingredients/IngredientsProductFragment.kt]: Could you invert this `if` and place it before `ingredientExtracted = true`?

**Review by VaiTon — CHANGES_REQUESTED** [review]

**Comment by mrudultora** [comment]
@VaiTon  Changes Done !

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/features/product/view/ingredients/IngredientsProductFragment.kt (hunk 1 above)
Comment (VaiTon) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/features/product/view/ingredients/IngredientsProductFragment.kt]:
```suggestion
        if (!isAdded) return
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/features/product/view/ingredients/IngredientsProductFragment.kt [hunk 2]
```diff
@@ -378,13 +378,15 @@ class IngredientsProductFragment : BaseFragment(), IIngredientsProductPresenter.
...
+            ingredientExtracted = true
+            val settings = requireActivity().getLoginPreferences()
+            if (settings.getString("user", "")!!.isEmpty()) {
+                showSignInDialog()
+            } else {
+                productState = requireProductState()
+                performOCRLauncher.launch(productState.product)
+            }
         }
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/features/product/view/ingredients/IngredientsProductFragment.kt]:
```suggestion
```

**Review by VaiTon — CHANGES_REQUESTED** [review]

**Pull request merged**